### PR TITLE
Add support for PKCS8 keys to Istio PKI

### PR DESCRIPTION
**Pull request by myidpt**

### CODE REVIEW

File: security/pkg/pki/crypto_test.go [hunk 1]
```diff
@@ -192,21 +231,21 @@ func TestParsePemEncodedKey(t *testing.T) {
...
 			keyType: reflect.TypeOf(&ecdsa.PrivateKey{}),
 		},
 		"Parse invalid ECDSA key": {
-			pem: `
------BEGIN EC PARAMETERS-----
------END EC PARAMETERS-----`,
+			pem:    keyInvalidECDSA,
 			errMsg: "failed to parse the ECDSA private key",
 		},
+		"Parse PKCS8 key using RSA algorithm": {
```
Comment (lei-tang) [inline security/pkg/pki/crypto_test.go]: In TestParsePemEncodedKey(), both RSA key and ECDSA key have two test cases, one for valid key and one for invalid key. To be consistent with the test cases of RSA key and ECDSA key, should PKCS8 key also have two test cases (one for a valid key and one for invalid key)?

Comment (myidpt) [reply security/pkg/pki/crypto_test.go]: Done.

**Comment by myidpt** [comment]
太痛苦。。LOL

**Pull request merged**